Петров Евгений

LeMarck opened the pull request:
Я сильно извиняюсь, у меня была проблема с отсутствием доступа к GitHub

LeMarck committed:
Add code

LeMarck committed:
Fix

LeMarck committed:
Add code

LeMarck committed:
Add codestyle;)

LeMarck committed:
ReCodeStyle

LeMarck committed:
ReCodeStyle

LeMarck committed:
Code style

LeMarck committed:
Code style

LeMarck committed:
Code style

Кодстайл - не моё))
Локальная проверка мне говорила что всё ОК

LeMarck committed:
Fix

Меняйте корректно

LeMarck committed:
Fix

LeMarck committed:
Code fix

Zhigalov commented on robbery.js:
```diff
@@ -2,14 +2,146 @@
+                if (times[day].from[1] < robber.to[1] &&
+                    robber.to[1] < times[day].to[1]) {
+                    times[day].from[1] = robber.to[1];
+                }
```
Может быть можно это както упростить используя `Math.min` или `Math.max`?

Zhigalov commented:
Если выбираю продолжительность ограбления 30 минут (или 3 часа), то получаю ошибку:

```
     TypeError: Cannot read property '1' of null
      at Object.module.exports.getAppropriateMoment (LeMarck/robbery.js:131:31)
      at Object.testCase (test.js:15:39)
```

Zhigalov commented:
Если время работы банка с 23 до 00, то получаю ошибку:

```
     TypeError: Cannot read property 'hour' of null
      at Object.format (LeMarck/moment.js:16:46)
      at Object.testCase (test.js:25:38)
```

LeMarck committed:
Upgrade

LeMarck committed:
Upgrade

LeMarck committed:
update codestyle

LeMarck committed:
update codestyle

LeMarck committed:
update_codestyle++

LeMarck committed:
added additional task

LeMarck committed:
added additional task

LeMarck committed:
re

LeMarck commented:
@Zhigalov, почему тесты не проходятся? Hang tight, the log cannot be shown until the build has started.

LeMarck committed:
Add codestyle;)

Zhigalov commented:
Если часы работы банка с 23 до 00 и на ограбление нужно час, то ожидаю `'Ограбление должно состоятся в СР. Всем быть готовыми к 23:00!'` а получаю `undefined`

Zhigalov commented on moment.js:
```diff
@@ -10,11 +12,34 @@ module.exports = function () {
         // Возвращает кол-во времени между текущей датой и переданной `moment`
         // в человекопонятном виде
         fromMoment: function (moment) {
+            var day = moment[0] != 0 ? moment[0] + ' дней ' : '';
```
Сложность дополнительного задания в том, что нужно правильно просклонять числительные, то есть вместо
`'До ограбления остался 1 минут'` нужно писать `'До ограбления осталась 1 минута'`

Zhigalov commented on robbery.js:
```diff
@@ -10,15 +75,93 @@ module.exports.getAppropriateMoment = function (json, minDuration, workingHours)
+                times[notFree.from[0]].to.push(notFree.to[1]);
+            } else {
+                times[notFree.from[0]].from.push(notFree.from[1]);
+                times[notFree.from[0]].to.push('23:59');
```
В нескольких местах у тебя повторяется `'23:59'`
Может быть стоит вынести это в константу/переменную?

Zhigalov commented:
Код написан хорошо, и большинство тестов проходит правильно.
Для того чтобы принять основное задание нужно поправить ошибку со временем работы банка с 23:00 до 00:00.
Для доп задания нужны склонения.

LeMarck committed:
update

Проблема была в том, что я день заканчиваю в 23:59 и тем самым теряю минуту до 00:00. Поправил :)

LeMarck committed:
update

Вот как знал, что склонять надо )))

LeMarck committed:
Add codestyle;)

LeMarck committed:
Add codestyle;)

Zhigalov commented:
По основной части задания - ок, по дополнительной:
- [ ] Вместо 'До ограбления остался 1 минута' ожидаю 'До ограбления осталась 1 минута'
- [ ] Вместо 'До ограбления остался 2 минуты' ожидаю 'До ограбления осталось 2 минуты'
- [ ] Вместо 'До ограбления остался 5 минут' ожидаю 'До ограбления осталось 5 минут'
- [ ] Вместо 'До ограбления остался 2 часа ' ожидаю 'До ограбления осталось 2 часа'

Zhigalov commented on moment.js:
```diff
@@ -1,5 +1,21 @@
 
+var WEEK = ['ПН', 'ВТ', 'СР'];
+
+function correct(num) {
```
Странное название функции))
Может лучше `plural`?

LeMarck committed:
update

LeMarck committed:
update

Zhigalov commented:
Отлично :star: Задача и доп приняты
